lib/Backend/GlobOpt.cpp

```diff
@@ -6417,6 +6420,38 @@ GlobOpt::GetConstantVar(IR::Opnd *opnd, Value *val)
     return nullptr;
 }
 
+Js::Var
+GlobOpt::GetConstantVar(IR::Opnd *opnd)
+{
+    if (opnd->IsAddrOpnd())
+    {
+        IR::AddrOpnd *addrOpnd = opnd->AsAddrOpnd();
+        if (addrOpnd->IsVar())
+        {
+            return addrOpnd->m_address;
+        }
+    }
+    else if (opnd->IsIntConstOpnd())
+    {
+        if (!Js::TaggedInt::IsOverflow(opnd->AsIntConstOpnd()->AsInt32()))
+        {
+            return Js::TaggedInt::ToVarUnchecked(opnd->AsIntConstOpnd()->AsInt32());
+        }
+    }
+    else if (opnd->IsRegOpnd() && opnd->AsRegOpnd()->m_sym->IsSingleDef())
+    {
+        IR::Instr * defInstr = opnd->AsRegOpnd()->m_sym->GetInstrDef();
+
+        if (defInstr->m_opcode != Js::OpCode::Ld_A)
+        {
+            return nullptr;
+        }
+        return GetConstantVar(defInstr->GetSrc1());
+    }
+
+    return nullptr;
+}
+
 bool BoolAndIntStaticAndTypeMismatch(Value* src1Val, Value* src2Val, Js::Var src1Var, Js::Var src2Var)
```

Comment:
>BoolAndIntStaticAndTypeMismatch [](start = 5, length = 31)

unused?

Reply:
Thanks for catching, it is an early exit check for a few branch types.